src/views/Visualizer/View/View.js

```diff
@@ -235,7 +241,30 @@ export class Visualizer extends React.Component {
         ...legendsCluster.reduce((acc, legend) => [...acc, { title, ...legend }], []),
       ], []);
 
-    return [...(legends || []), ...(legendsFromLayersTree || [])];
+    const allLegends = [...(legends || []), ...(legendsFromLayersTree || [])];
```

Comment:
It might be more readable to set the empty array as the default property of `legends` line 220.
Also, I don't think it's necessary to set an empty array as the fallback of `legendsFromLayersTree` because `Array.from` above should return at least an empty array.

Reply:
Completely agree, but as I didn't really change logic of this in this PR, I might do change suggested in a later another PR.